java/core/src/main/java/com/microsoft/bond/NullableBondType.java

```diff
@@ -120,6 +120,20 @@ protected final TValue deserializeValue(TaggedDeserializationContext context) th
     }
 
     @Override
+    protected final TValue deserializeValue(UntaggedDeserializationContext context) throws IOException {
+        final int count = context.reader.readContainerBegin();
+        if (count == 0) {
+            return null;
+        } else if (count == 1) {
```

Comment:
It probably should. I don't think omitting fields is ever mandatory, so we need to be able to handle nullables that come as explicit lists of length 0.